### ISSUE TITLE
CA1416 emits warnings for internal code

### ISSUE THREAD

**Issue by bgavrilMS** [dotnet/roslyn-analyzers#4933]
### Analyzer

**Diagnostic ID**: CA1416 

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 5.0.200

### Repro

```xml
    <TargetFramework>net5.0-windows10.0.17763.0</TargetFramework>
    <SupportedOSPlatformVersion>7</SupportedOSPlatformVersion>
```
```csharp

    internal class Stuff 
    {
        public void Foo() {           // logic that calls into WinRT }
    }
```

Expected: no warning since code is internal. 
Actual: CA1416 warning

Note: We (MSAL.NET) does not maintain internal API surface and change it frequently.

**Comment by Youssef1313** [comment]
@buyaa-n Can you take a look whether this is indeed a false positive or not?

**Comment by buyaa-n** [comment]
> Expected: no warning since code is internal.

Not sure what you meant by internal and why you think it should not warn, JFYI: platform compatibility warning not depend on API surface level

I guess you meant the project is targeting windows 10 but supports windows 7 and above, so calling windows only `Foo()` should not warn. As you didn't show the warning message i cannot really say what was wrong with that call. But i guess `7` might not be counted as a valid version, I suggest using `7.0` instead

**Comment by bgavrilMS** [comment]
Got it, I did not realize that API surface level does not influence platform compat warning. I think it makes sense. Closing.
